### DISCUSSION TITLE
Pre render for quarto preview

### DISCUSSION THREAD

**Discussion by itepifanio** [quarto-dev/quarto-cli#4822] [Feature Requests]
Quarto provides [pre and pos rendering](https://quarto.org/docs/projects/scripts.html#pre-and-post-render) features but it doesn't extend when live previewing the documents.

I have a script that reads `yaml` content and outputs `.qmd` files used in a Quarto website. I was trying to add a custom watcher for the `yaml` files, this way, when a `yaml` file changes the `.qmd` files the Quarto live previewing would refresh. 

My solution was running Quarto as a subprocess and creating the custom watcher. A pre rendering that works with live previewing would make my life easier because I would be able to run the `yaml` watcher script without working with subprocessing.

**Comment by mcanouil** [comment]
Closing in favour of https://github.com/quarto-dev/quarto-cli/issues/4841
